src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (C) 2019 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <climits>
+#include <getopt.h>
+#include <dirent.h>
+#include <string.h>
+
+#include <boost/filesystem.hpp>
+
+#include "xbmgmt.h"
+#include "core/pcie/linux/scan.h"
+
+using namespace boost::filesystem;
+
+#define SYSFS_PATH      "/sys/bus/pci/devices"
+#define XILINX_VENDOR   "0x10ee"
+#define XILINX_US       "0x9134"
+
+static std::string findRootPort(void);
+static int removeDevice(std::shared_ptr<pcidev::pci_device> dev);
+
+const char *subCmdHpRemoveDesc = "Perform managed hot remove on the device";
+const char *subCmdHpRemoveUsage = "--card b:d.f";
+
+int hpRemoveHandler(int argc, char *argv[])
+{
+    sudoOrDie();
+
+    unsigned index = UINT_MAX;
+    const option opts[] = {
+        { "card", required_argument, nullptr, '0' },
+    };
+
+    if (argc != 3)
+        return -EINVAL;
+
+    while (true) 
+    {
+        const auto opt = getopt_long(argc, argv, "", opts, nullptr);
+        if (opt == -1)
+            break;
+
+        switch (opt) 
+        {
+            case '0':
+		char ch;
+                index = bdf2index(optarg);
+                if (index == UINT_MAX)
+                    return -ENOENT;
+            	std::cout << "CAUTION: Performing hot removal. This command is going to impact both user pf and mgmt pf. " <<
+                "Please make sure no application is currently running." << std::endl;
+		std::cout << "Proceed[y/n]? " << std::endl;
+		std::cin >> ch;
+		if (ch == 'y' || ch == 'Y')
+			break;
+		else
+			return -EINVAL;
+
+            default:
+                return -EINVAL;
+        } 
+    }
+
+    /* Remove user_pf */
+    auto uDev = pcidev::get_dev(index, true);
+    removeDevice(uDev);
+
+    /* Remove mgmt_pf */
+    auto mDev = pcidev::get_dev(index, false);
+    removeDevice(mDev);
+    
+    return 0;
+}
+
+static int removeDevice(std::shared_ptr<pcidev::pci_device> dev)
```

Comment:
should just pass in the raw pointer to pcidev::pci_device here.

Reply:
The previous function where we are getting pci_device structure is returning a shared ptr.  Hence used the same. Type casing is not working for me. 
Please check and suggest if this is really required.